app/models/tag.rb

```diff
@@ -1023,7 +1033,7 @@ def remove_sub_filters(sub_tag)
   before_update :check_canonical
   def check_canonical
     if self.canonical_changed? && !self.canonical?
-      self.async(:remove_canonical_associations)
+      remove_canonical_associations
```

Comment:
Looking at the two, I still think remove_canonical_associations might take a while.
Why does the async make any difference to the tests as I thought all async calls were synchronous in test ?